### PR TITLE
Add documentation for ICorDebugProcess4

### PR DESCRIPTION
**Pull request by hoyosjs**
Adds documentation for ICorDebugProcess 4 to complete Process debugging pipelines.

### CODE REVIEW

File: docs/framework/unmanaged-api/debugging/cordebugstatechange-enumeration.md [hunk 1]
```diff
@@ -1,53 +1,50 @@
...
+
+```
+typedef enum CorDebugStateChange
+{
+    PROCESS_RUNNING = 0x0000001,
+    FLUSH_ALL       = 0x0000002,
+} CorDebugStateChange;
+```
+
+## Members
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/cordebugstatechange-enumeration.md]: can you update the date on line 3 to 02/08/2019?

File: docs/framework/unmanaged-api/debugging/debugging-interfaces.md [hunk 2]
```diff
@@ -296,7 +296,10 @@ This section describes the unmanaged interfaces that handle the debugging of a p
  A logical extension of `ICorDebugProcess`.  
   
  [ICorDebugProcess3 Interface](../../../../docs/framework/unmanaged-api/debugging/icordebugprocess3-interface.md)
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/debugging-interfaces.md]: same here, update ms.date

File: docs/framework/unmanaged-api/debugging/debugging-interfaces.md [hunk 3]
```diff
@@ -296,7 +296,10 @@ This section describes the unmanaged interfaces that handle the debugging of a p
  A logical extension of `ICorDebugProcess`.  
   
  [ICorDebugProcess3 Interface](../../../../docs/framework/unmanaged-api/debugging/icordebugprocess3-interface.md)  
- Controls custom debugger notifications.  
+ Controls custom debugger notifications.
+
+ [ICorDebugProcess4 Interface](../../../../docs/framework/unmanaged-api/debugging/icordebugprocess4-interface.md)
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/debugging-interfaces.md]:
```suggestion
 [ICorDebugProcess4 Interface](../../../../docs/framework/unmanaged-api/debugging/icordebugprocess4-interface.md)\
```

You need either the slash at the end or the two white spaces to break the line. We're moving away from relying on the two spaces at the end

image

File: docs/framework/unmanaged-api/debugging/debugging-interfaces.md [hunk 4]
```diff
@@ -296,7 +296,10 @@ This section describes the unmanaged interfaces that handle the debugging of a p
...
+ Provides support for out of process execution control.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/debugging-interfaces.md]: the same problem occurs for the other new APIs that were recently added at the bottom of this article

File: docs/framework/unmanaged-api/debugging/icordebugprocess4-interface.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ms.author: "juhoyosa"
+---
+# ICorDebugProcess4 Interface
+Provides support for out of process execution control.
+
+## Methods
+
+| Method                                                                                                                    | Description                                                                                             |
+| ------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------- |
+| [ProcessStateChanged](../../../../docs/framework/unmanaged-api/debugging/icordebugprocess4-processstatechanged-method.md) | Notifies the ICorDebug pipeline that the out of process debugger is continuing the debugee's execution. |
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/icordebugprocess4-interface.md]: Easier to use the shorter path. The long paths are due to the script that migrated the content from MSDN

```suggestion
| [ProcessStateChanged](icordebugprocess4-processstatechanged-method.md) | Notifies the ICorDebug pipeline that the out of process debugger is continuing the debugee's execution. |
```

File: docs/framework/unmanaged-api/debugging/icordebugprocess4-interface.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+## Remarks
+
+This interface lives inside the runtime and is not exposed through any headers or library files. However, it's a COM interface that derives from `IUnknown` with GUID `E930C679-78AF-4953-8AB7-B0AABF0F9F80` that can be obtained through the usual COM mechanisms.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/icordebugprocess4-interface.md]:
```suggestion
This interface lives inside the runtime and isn't exposed through any headers or library files. However, it's a COM interface that derives from `IUnknown` with GUID `E930C679-78AF-4953-8AB7-B0AABF0F9F80` that can be obtained through the usual COM mechanisms.
```

File: docs/framework/unmanaged-api/debugging/icordebugprocess4-processstatechanged-method.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+```
+HRESULT ProcessStateChanged(
+    [in] CorDebugStateChange change
+);
+```
+
+#### Parameters
+ `eChange`
+ [in] A member of the [CorDebugStateChange enumeration](../../../../docs/framework/unmanaged-api/debugging/cordebugstatechange-enumeration.md) describing a change in the process's execution state.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/icordebugprocess4-processstatechanged-method.md]:
```suggestion
 [in] A member of the [CorDebugStateChange enumeration](cordebugstatechange-enumeration.md) describing a change in the process's execution state.
```

File: docs/framework/unmanaged-api/debugging/icordebugprocess4-processstatechanged-method.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+## Syntax
+
+```
+HRESULT ProcessStateChanged(
+    [in] CorDebugStateChange change
+);
+```
+
+#### Parameters
+ `eChange`
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/icordebugprocess4-processstatechanged-method.md]:
```suggestion

 `eChange`\
```

File: docs/framework/unmanaged-api/debugging/icordebugprocess4-processstatechanged-method.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  - "ICorDebugProcess4::ProcessStateChanged"
+helpviewer_keywords:
+  - "ICorDebugProcess4::ProcessStateChanged method [.NET Framework debugging]"
+topic_type:
+  - "apiref"
+author: "hoyosjs"
+ms.author: "juhoyosa"
+---
+# ICorDebugProcess4::ProcessStateChanged Method
+Notifies the ICorDebug pipeline that the out of process debugger is continuing the debugee's execution.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/icordebugprocess4-processstatechanged-method.md]:
```suggestion

Notifies the ICorDebug pipeline that the out of process debugger is continuing the debugee's execution.
```

File: docs/framework/unmanaged-api/debugging/icordebugprocess4-interface.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+  - "ICorDebugProcess4"
+helpviewer_keywords:
+  - "ICorDebugProcess4 interface [.NET Framework debugging]"
+topic_type:
+  - "apiref"
+author: "hoyosjs"
+ms.author: "juhoyosa"
+---
+# ICorDebugProcess4 Interface
+Provides support for out of process execution control.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/icordebugprocess4-interface.md]:
```suggestion

Provides support for out of process execution control.
```

File: docs/framework/unmanaged-api/debugging/cordebugstatechange-enumeration.md [hunk 11]
```diff
@@ -1,53 +1,50 @@
...
+ A member of the `CorDebugStateChange` enumeration is provided as an argument when the debugger calls the `ProcessStateChanged` method either with [ICorDebugProcess4::ProcessStateChanged](../../../../docs/framework/unmanaged-api/debugging/icordebugprocess4-processstatechanged-method.md) or [ICorDebugProcess6::ProcessStateChanged](../../../../docs/framework/unmanaged-api/debugging/icordebugprocess6-processstatechanged-method.md)
+
+## Requirements
+ **Platforms:** See [System Requirements](../../../../docs/framework/get-started/system-requirements.md).
+
+ **Header:** CorDebug.idl, CorDebug.h
+
+ **Library:** CorGuids.lib
+
+ **.NET Framework Versions:** [!INCLUDE[net_46_native](../../../../includes/net-46-native-md.md)]  and [!INCLUDE[net_current_v45plus](../../../../includes/net-current-v20plus-md.md)]
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/cordebugstatechange-enumeration.md]: This doesn't seem right

image

File: docs/framework/unmanaged-api/debugging/icordebugprocess4-processstatechanged-method.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+);
+```
+
+#### Parameters
+ `eChange`
+ [in] A member of the [CorDebugStateChange enumeration](../../../../docs/framework/unmanaged-api/debugging/cordebugstatechange-enumeration.md) describing a change in the process's execution state.
+
+## Remarks
+
+The provided method is part of the `ICorDebugProcess4` interface and corresponds to the 4th slot of the virtual method table.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/debugging/icordebugprocess4-processstatechanged-method.md]:
```suggestion
The provided method is part of the `ICorDebugProcess4` interface and corresponds to the fourth slot of the virtual method table.
```

**Review by mairaw — DISMISSED** [review]
Looks great @hoyosjs. Left a few comments/suggestions for you to consider.

**Review by hoyosjs — COMMENTED** [review]
Thanks @mairaw! I think I've finished addressing the feedback. I've also cleaned up the debugging interfaces index to use the `\` convention as well as the short link one, but if the change is too large I can revert it.

**Comment by hoyosjs** [comment]
Sorry to nudge @rpetrusha (or @mairaw), but there's a release coming out tomorrow that depends on this documentation. Is there anything else here that should get cleaned up? Thanks!

**Review by rpetrusha — APPROVED** [review]
LGTM. I'll merge when the build completes, @hoyosjs.

**Comment by hoyosjs** [comment]
Thanks!

**Pull request merged**